src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/FlowMirrorPathFrame.java

```diff
@@ -0,0 +1,229 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.ferma.frames;
+
+import org.openkilda.model.FlowMirrorPath.FlowMirrorPathData;
+import org.openkilda.model.FlowMirrorPoints;
+import org.openkilda.model.FlowPathStatus;
+import org.openkilda.model.PathId;
+import org.openkilda.model.PathSegment;
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchId;
+import org.openkilda.model.cookie.FlowSegmentCookie;
+import org.openkilda.persistence.ferma.frames.converters.Convert;
+import org.openkilda.persistence.ferma.frames.converters.FlowPathStatusConverter;
+import org.openkilda.persistence.ferma.frames.converters.FlowSegmentCookieConverter;
+import org.openkilda.persistence.ferma.frames.converters.PathIdConverter;
+import org.openkilda.persistence.ferma.frames.converters.SwitchIdConverter;
+
+import com.syncleus.ferma.annotations.Property;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.tinkerpop.gremlin.structure.Direction;
+import org.apache.tinkerpop.gremlin.structure.Edge;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+public abstract class FlowMirrorPathFrame extends KildaBaseVertexFrame implements FlowMirrorPathData {
+    public static final String FRAME_LABEL = "flow_mirror_path";
+    public static final String OWNS_SEGMENTS_EDGE = "owns";
+    public static final String PATH_ID_PROPERTY = "path_id";
+    public static final String SRC_SWITCH_ID_PROPERTY = "src_switch_id";
+    public static final String DST_SWITCH_ID_PROPERTY = "dst_switch_id";
+    public static final String DST_PORT_PROPERTY = "dst_port";
+    public static final String DST_VLAN_PROPERTY = "dst_vlan";
+    public static final String COOKIE_PROPERTY = "cookie";
+    public static final String IGNORE_BANDWIDTH_PROPERTY = "ignore_bandwidth";
+    public static final String BANDWIDTH_PROPERTY = "bandwidth";
+    public static final String DST_MULTI_TABLE_PROPERTY = "dst_with_multi_table";
+
+    private Switch srcSwitch;
```

Comment:
This is the switch where we make a mirror? Can we name it `mirrorSwitch`?

Reply:
Renamed.